src/test/kotlin/no/nav/tilleggsstonader/sak/utbetaling/iverksetting/IverksettStatusServiceTest.kt

```diff
@@ -0,0 +1,115 @@
+package no.nav.tilleggsstonader.sak.utbetaling.iverksetting
+
+import io.mockk.every
+import no.nav.familie.prosessering.error.TaskExceptionUtenStackTrace
+import no.nav.tilleggsstonader.sak.IntegrationTest
+import no.nav.tilleggsstonader.sak.behandling.domain.Behandling
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingResultat
+import no.nav.tilleggsstonader.sak.infrastruktur.database.repository.findByIdOrThrow
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.Feil
+import no.nav.tilleggsstonader.sak.infrastruktur.mocks.IverksettClientConfig.Companion.clearMock
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.TilkjentYtelseUtil.andelTilkjentYtelse
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.TilkjentYtelseUtil.tilkjentYtelse
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.AndelTilkjentYtelse
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.Iverksetting
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.StatusIverksetting
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.TilkjentYtelseRepository
+import no.nav.tilleggsstonader.sak.util.behandling
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import org.springframework.beans.factory.annotation.Autowired
+import java.time.LocalDateTime
+
+class IverksettStatusServiceTest : IntegrationTest() {
+
+    @Autowired
+    lateinit var iverksettClient: IverksettClient
+
+    @Autowired
+    lateinit var iverksettStatusService: IverksettStatusService
+
+    @Autowired
+    lateinit var tilkjentYtelseRepository: TilkjentYtelseRepository
+
+    private val behandling = behandling(resultat = BehandlingResultat.INNVILGET)
+
+    @BeforeEach
+    fun setUp() {
+        clearMock(iverksettClient)
+        testoppsettService.opprettBehandlingMedFagsak(behandling)
+    }
+
+    @AfterEach
+    override fun tearDown() {
+        super.tearDown()
+        clearMock(iverksettClient)
+    }
+
+    @Test
+    fun `skal oppdatere iverksatte andeler med status OK`() {
+        val iverksattAndel = andelTilkjentYtelse(
+            kildeBehandlingId = behandling.id,
+            statusIverksetting = StatusIverksetting.SENDT,
+            iverksetting = Iverksetting(behandling.id, LocalDateTime.now()),
+        )
+        val andelIkkeSendt = andelTilkjentYtelse(kildeBehandlingId = behandling.id)
+        val tilkjentYtelse = opprettTilkjentYtelse(behandling, iverksattAndel, andelIkkeSendt)
+
+        iverksettStatusService.hentStatusOgOppdaterAndeler(behandling.id, behandling.id)
```

Comment:
Hvorfor feiler ikke denne når en av andelen har stattus ikkeSendt?

Reply:
Andeler for status SENDT og iverksettingId når de faktiskt har blitt iverksatte

https://github.com/navikt/tilleggsstonader-sak/pull/191/files#diff-ca79609f7e76db879eaf1fa0d96cfe5e6f6ec36afb1ebb9fe952d646158dd800R40-R46

Selve `hentStatusOgOppdaterAndeler` mottar behandlingId og iverksettingId
Den plukker ut andeler med gitt iverksettingId
Hvis noen av de andelene ikke har status SENDT skal det feile
Hvis det ikke finnes noen andeler av gitt iverksettingId skal det feile